src/common/utils.cpp

```diff
@@ -67,6 +67,24 @@ void split(const std::string& line,
   }
 }
 
+void splitTsv(const std::string& line, std::vector<std::string>& fields, size_t numFields) {
+  fields.clear();
+  if(fields.capacity() != numFields) // make sure the number of fields is always as requested
```

Comment:
Thanks! I changed that to resizing the vector at the end of the function, to guarantee that the output has as many elements as requested. I am not sure if it's actually necessary, but I think it's safer.

Added also a regression test with excessive tabs and simple unit tests.

Reply:
Somehow I cannot see the changed code in here, but let's make it a hard error. Either the file contains the correct number of TAB-separated fields, or it is malformed, and the training should terminate.

We could consider skipping a line with a warning, but I suggest to make it a hard error for now until someone shouts about it.

Reply:
GitHub is not good at presenting changes for a review. It's here: https://github.com/marian-nmt/marian-dev/blob/6da35a2d8dd9f9a2db503d4bb954d72b3a8a9c01/src/common/utils.cpp#L70-L91

I think that a hard error might be too much. I would be annoyed if my job waiting for two days in a slurm queue is aborted because only of a few malformed lines in the corpus. If there is many malformed lines, the logging file will be spammed with warning messages, for example: https://github.com/marian-nmt/marian-regression-tests/blob/325fd45d0a8a0648c39cc26e9757b8bee246d20d/tests/interface/input-tsv/train_extra_tabs.expected

At the moment there is a warning and the line is not skipped, but I now think that it should be.

Reply:
My `StringPiece` / `string_ref` tic is triggered by this function declaration.

Reply:
Also, I vote for hard error.

Reply:
We have warnings for other failed checks, e.g. a missing sentence weight, missing or excessive word weights. Maybe these should be hard errors too? OK, if I'm not wrong, incorrect word alignment during guided alignment is a segmentation fault, so it's actually a hard error...

Reply:
Replaced the warning with an abort.